Tests/DatadogTests/Datadog/TracerTests.swift

```diff
@@ -333,9 +333,11 @@ class TracerTests: XCTestCase {
         let tracer = Tracer.initialize(configuration: .init()).dd
 
         tracer.startSpan(operationName: "span with no user info").finish()
+        tracer.queue.sync {} // wait for processing the span event in `DDSpan`
```

Comment:
💡  We need to synchronize test execution like this, otherwise the `Span` creation asynchronously dispatched from may fall into race condition `.setUserInfo()` called in the next line. It's a common trick, already used widely [in `RUMMonitorTests`](https://github.com/DataDog/dd-sdk-ios/blob/1cd1a46c61e6a792e138b3aac00a29f5a12c97e7/Tests/DatadogTests/Datadog/RUMMonitorTests.swift#L804).